ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Background.hs

```diff
@@ -271,17 +276,21 @@ copyAndSnapshotRunner cdb@CDB{..} gcSchedule replayed =
       scheduleGC' immSlotno
 
       -- Let the LedgerDB flush
-      LgrDB.withWriteLock cdbLgrDB $ LgrDB.flush cdbLgrDB
+      ctr <- readMVar flushCtr
+      if ctr + numToWrite > 100 then do
+        LgrDB.withWriteLock cdbLgrDB $ LgrDB.flush cdbLgrDB
+        putMVar flushCtr 0
+      else putMVar flushCtr (ctr + numToWrite)
```

Comment:
Won't this `putMVar` block forever as the `MVar` is never cleared? Obvious fixes would be to use `takeMVar` instead of `readMVar`, or switching to `IORef`.

(Just randomly noticed this as you mentioned this exact same thing earlier today.)